.github/workflows/main_ci.yml

```diff
@@ -34,190 +34,139 @@ jobs:
+        key: ${{ env.CACHE_VERSION }}-${{ env.CACHE_NAME }}-${{ matrix.os }}-${{ hashFiles('vcpkg_commit.txt', 'alternatives/*/vcpkg.json') }}
         restore-keys: |
           ${{ env.CACHE_VERSION }}-${{ env.CACHE_NAME }}-${{ matrix.os }}
 
-    - name: Dependencies (macOs)
-      if: ${{ matrix.os == 'macos-latest' }}
-      run: |
-        brew install llvm pkg-config nasm
-        ln -s "/usr/local/opt/llvm/bin/clang-format" "/usr/local/bin/clang-format"
-        ln -s "/usr/local/opt/llvm/bin/clang-tidy" "/usr/local/bin/clang-tidy"
-
     - name: Dependencies (Ubuntu)
       if: ${{ matrix.os == 'ubuntu-latest' }}
       run: |
         sudo apt-get install -y linux-headers-$(uname -r) nasm
 
-    - name: Build (OpenSSL 1.1)
-      run: |
-        cmake -B "${{ env.CMAKE_BUILD_DIR }}" -DTESTING=ON -DCLANG_TIDY=ON -DSANITIZERS=ON -DCMAKE_TOOLCHAIN_FILE="${{ env.VCPKG_TOOLCHAIN_FILE }}"
-        cmake --build "${{ env.CMAKE_BUILD_DIR }}"
-
-    - name: Unit Test (OpenSSL 1.1)
+    - name: Build
       run: |
-        cmake --build "${{ env.CMAKE_BUILD_DIR }}" --target "${{ matrix.ctest-target}}"
+        cmake -B "${{ env.BUILD_DIR }}" -DVCPKG_MANIFEST_DIR="alternatives/${{ matrix.crypto }}"
+        cmake --build "${{ env.BUILD_DIR }}"
 
-    - name: Build (OpenSSL 3)
+    - name: Build (Interop Harness)
       run: |
-        cmake -B "${{ env.CMAKE_BUILD_OPENSSL3_DIR }}" -DTESTING=ON -DCLANG_TIDY=ON -DSANITIZERS=ON -DVCPKG_MANIFEST_DIR="${{ matrix.ossl3-vcpkg-dir }}" -DCMAKE_TOOLCHAIN_FILE="${{ env.VCPKG_TOOLCHAIN_FILE }}"
-        cmake --build "${{ env.CMAKE_BUILD_OPENSSL3_DIR }}"
+        cd cmd/interop
+        cmake -B build
+        cmake --build build
 
-    - name: Unit Test (OpenSSL 3)
+    - name: Test self-interop
       run: |
-        cmake --build "${{ env.CMAKE_BUILD_OPENSSL3_DIR }}" --target "${{ matrix.ctest-target}}"
+        make -C cmd/interop self-test
 
-    - name: Build (BoringSSL)
+    - name: Test interop on test vectors
       run: |
-        cmake -B "${{ env.CMAKE_BUILD_BORINGSSL_DIR }}" -DTESTING=ON -DCLANG_TIDY=ON -DSANITIZERS=ON -DVCPKG_MANIFEST_DIR="${{ matrix.boringssl-vcpkg-dir }}" -DCMAKE_TOOLCHAIN_FILE="${{ env.VCPKG_TOOLCHAIN_FILE }}"
-        cmake --build "${{ env.CMAKE_BUILD_BORINGSSL_DIR }}"
+        make -C cmd/interop interop-test
 
-    - name: Unit Test (BoringSSL)
+    - name: Test gRPC live interop with self
       run: |
-        cmake --build "${{ env.CMAKE_BUILD_BORINGSSL_DIR }}" --target "${{ matrix.ctest-target}}"
+        cd cmd/interop
+        ./grpc-self-test.sh
 
-  old-macos-compatibility:
+  clang-tidy:
     if: github.event.pull_request.draft == false
-    needs: quick-linux-interop-check
-    name: Build for older MacOS
-    runs-on: macos-latest
+    needs: build-and-unit-test
+    name: Build with clang-tidy
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        crypto: [openssl_1.1, openssl_3, boringssl]
```

Comment:
It might clean things up to make this the vcpkg manifest directory instead of crypto.  It would make it more generic.  Maybe it is a premature optimization.  Just wanted to mention.